mock-kubernetes-client/client/client_test.go

```diff
@@ -153,6 +153,37 @@ func TestDeletingObject(t *testing.T) {
 	g.Expect(k8serrors.IsNotFound(err)).To(BeTrue())
 }
 
+func TestDeletingPodInTerminating(t *testing.T) {
+	g := NewWithT(t)
+	client := &MockClient{}
+
+	pod1 := createDummyPod()
+	pod2 := createDummyPod()
```

Comment:
What is the reason to have `pod2`?

Reply:
I wanted to add a test that when one pod is stuck in terminating, another pod can still terminate successfully.